src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -86,6 +88,66 @@ def best_distribution(planned_partitions, disk_names)
 
         calculator.best_distribution(planned_partitions, spaces)
       end
+
+    private
+
+      # Creates planned partitions in the given devicegraph
+      #
+      # @param planned_partitions [Array<Planned::Partition>] Devices to create/reuse
+      # @param disk_names [Array<String>] Disks to consider
+      # @return [PartitionCreatorResult]
+      def create_partitions(planned_partitions, disk_names)
+        log.info "Partitions to create: #{planned_partitions}"
+
+        dist = best_distribution(planned_partitions, disk_names)
+        raise Error, "Could not find a valid partitioning distribution" if dist.nil?
+        part_creator = Proposal::PartitionCreator.new(original_graph)
+        part_creator.create_partitions(dist)
+      end
+
+      # Creates volume groups in the given devicegraph
+      #
+      # @param vgs         [Array<Planned::LvmVg>]           List of planned volume groups to add
+      # @param devicegraph [Devicegraph]                     Starting point
+      # @param devices_map [Hash<String,Planned::Partition>] Map of device names and partitions
+      # @return [Devicegraph] Devicegraph containing the specified volume groups
+      def create_volume_groups(vgs, devicegraph, devices_map)
```

Comment:
Nitpicking: strictly speaking, this is not necessarily creating VGs. I would say the more important part here (that may not be obvious from the name) is that it's creating LVs. At least, not all the VGs included in the first argument are finally created, some of them are already there (reuse) and are simply "completed" with more LVs...

Nevermind, I'm probably thinking too much ;-)